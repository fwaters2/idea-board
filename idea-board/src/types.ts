export interface Idea {
  id: string;
  title: string;
  description: string;
  created: Date;
  updated?: Date;
}

export enum SortDirection {
  ASC = "ASC",
  DESC = "DESC",
}

export enum SortBy {
  CREATED = "CREATED",
  TITLE = "TITLE",
}

export enum Status {
  SUCCESS = "SUCCESS",
  ERROR = "ERROR",
  IDLE = "IDLE",
  LOADING = "LOADING",
}
