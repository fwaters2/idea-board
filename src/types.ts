export interface Idea {
  id: string;
  title: string;
  description: string;
  created: Date;
  updated?: Date;
}

export interface AppState {
  ideas: Idea[];
  sortKey: SortKey;
  sortDirection: SortDirection;
}

export type SortDirection = "asc" | "desc";

export type SortKey = "created" | "title";

export type Status = "success" | "error" | "idle" | "loading";
