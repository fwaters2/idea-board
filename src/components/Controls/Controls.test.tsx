import React from "react";
import { SortKey, SortDirection } from "../../types";
import { Controls } from ".";
import { render } from "@testing-library/react";

jest.mock("../NewCardBtn", () => ({
  NewCardBtn: () => <div>NewCardBtn</div>,
}));

describe("Controls component", () => {
  const handleAddCard = jest.fn();
  const onClickSort = jest.fn();
  const props = {
    handleAddCard,
    onClickSort,
    sortDirection: "asc" as SortDirection,
    sortKey: "created" as SortKey,
  };

  afterEach(() => {
    handleAddCard.mockClear();
    onClickSort.mockClear();
  });

  test("should render correctly", () => {
    const { container } = render(<Controls {...props} />);
    expect(container).toMatchSnapshot();
  });
});
