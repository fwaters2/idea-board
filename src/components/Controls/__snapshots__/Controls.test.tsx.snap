// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component should render correctly 1`] = `
<div>
  <div
    class="controls-container"
  >
    <div>
      NewCardBtn
    </div>
    <button
      class="control_btn"
    >
      <span>
        ↓
      </span>
      <span>
        Created
      </span>
    </button>
    <button
      class="control_btn"
    >
      <span>
        ⇅
      </span>
      <span>
        Title
      </span>
    </button>
  </div>
</div>
`;
